In [36]:
import pandas as pd
import numpy as np

In [37]:
df = pd.read_csv('Datasets/Combination.csv')

In [38]:
df.head()

N     P    K    pH    EC    OC    Zn   Fe    Cu   Mn  Output
0   75  20.0  279  7.74  0.40  0.01  0.48  6.4  0.21  4.7       1
1   85  15.7  247  9.02  0.31  0.02  0.27  6.4  0.16  5.6       1
2   77  35.6  265  7.80  0.17  0.02  0.46  6.2  0.51  6.1       1
3  106   6.4  127  8.36  0.02  0.03  0.50  3.1  0.28  2.3       0
4   96  10.5   96  8.36  1.08  0.03  0.31  3.2  0.23  4.1       0

In [39]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 980 entries, 0 to 979
Data columns (total 11 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   N       980 non-null    int64  
 1   P       980 non-null    float64
 2   K       980 non-null    int64  
 3   pH      980 non-null    float64
 4   EC      980 non-null    float64
 5   OC      980 non-null    float64
 6   Zn      980 non-null    float64
 7   Fe      980 non-null    float64
 8   Cu      980 non-null    float64
 9   Mn      980 non-null    float64
 10  Output  980 non-null    int64  
dtypes: float64(8), int64(3)
memory usage: 84.3 KB


In [40]:
df.shape

(980, 11)

Renaming Output to fertility

In [41]:
df = df.rename(columns={'Output': 'Fertility'})

In [42]:
df.head()

N     P    K    pH    EC    OC    Zn   Fe    Cu   Mn  Fertility
0   75  20.0  279  7.74  0.40  0.01  0.48  6.4  0.21  4.7          1
1   85  15.7  247  9.02  0.31  0.02  0.27  6.4  0.16  5.6          1
2   77  35.6  265  7.80  0.17  0.02  0.46  6.2  0.51  6.1          1
3  106   6.4  127  8.36  0.02  0.03  0.50  3.1  0.28  2.3          0
4   96  10.5   96  8.36  1.08  0.03  0.31  3.2  0.23  4.1          0

Checking the shape of fertile and non fertile data points

In [43]:
df[df['Fertility'] == 1].shape

(529, 11)

In [44]:
df[df['Fertility'] == 0].shape

(451, 11)

Dividing them accordingly

In [45]:
fertile = df[df['Fertility'] == 1]
nonfertile = df[df['Fertility'] == 0]

In [46]:
fertile.head()

N     P    K    pH    EC    OC    Zn   Fe    Cu   Mn  Fertility
0   75  20.0  279  7.74  0.40  0.01  0.48  6.4  0.21  4.7          1
1   85  15.7  247  9.02  0.31  0.02  0.27  6.4  0.16  5.6          1
2   77  35.6  265  7.80  0.17  0.02  0.46  6.2  0.51  6.1          1
8  112  27.0  333  7.87  0.43  0.04  0.75  3.9  0.54  1.8          1
9   89  28.2  146  8.09  0.62  0.04  0.53  3.8  0.71  3.9          1

In [47]:
nonfertile.head()

N     P    K    pH    EC    OC    Zn   Fe    Cu   Mn  Fertility
3  106   6.4  127  8.36  0.02  0.03  0.50  3.1  0.28  2.3          0
4   96  10.5   96  8.36  1.08  0.03  0.31  3.2  0.23  4.1          0
5  151  10.5  230  8.36  0.73  0.03  0.38  2.5  0.37  4.2          0
6  112   8.0  120  7.69  0.11  0.04  0.51  3.1  0.32  1.2          0
7  125  18.5  145  8.39  0.06  0.04  0.67  2.8  0.18  1.8          0

Randomly sampling 500 rows from the subset of fertile as we have 529 right now

In [48]:
fertile_sampled = fertile.sample(n=500, random_state=42)

In [49]:
fertile_sampled.shape

(500, 11)

Randomly sampling 400 rows from the subset of nonfertile as we have 451 right now

In [50]:
nonfertile_sampled = nonfertile.sample(n=300, random_state=43)

In [51]:
nonfertile_sampled.shape

(300, 11)

Making a new Dataset with 500 fertile and 300 nonfertile data points

In [53]:
df_main = pd.concat([fertile_sampled, nonfertile_sampled])

In [54]:
df_main.head()

N     P    K    pH    EC    OC    Zn    Fe    Cu     Mn  Fertility
400  370   9.4  729  7.30  0.59  0.10  0.51  8.05  0.53  11.25          1
828  339   6.8  570  7.50  0.79  0.59  0.13  1.45  0.33   2.72          1
12    78   8.4  190  8.12  0.14  0.04  0.62  1.00  0.07   5.50          1
754  264   9.9  454  7.40  0.33  0.73  0.42  0.81  0.23   4.00          1
721  207  76.8  560  7.42  0.42  0.78  0.48  4.15  1.02  13.25          1

In [56]:
df_main.shape

(800, 11)

Randomizing the fertile and non fertile data points

In [58]:
df_main = df_main.sample(frac=1, random_state=42)

Saving the new dataset which we will use further for our project

In [59]:
df_main.to_csv('main_data.csv',index=False)